src/content/tools/hooks.md

```diff
@@ -0,0 +1,377 @@
+
+1. Discover your hooks.
+1. Execute your hooks with the necessary input.
+1. Consume the output produced by the hooks.
+
+Example project:
+
+```plaintext
+📁  example_project                   // project with the hooks
+    📁  build                         // assets are generated here
+        📄 asset
+        📄 asset
+        ...
+    📁  hook                          // add hook scripts here
+        📄 build.dart
+        📄 link.dart
+    📁  lib                           // reference your assets here
+        📄 example.dart
+    📁  src                           // add native assets here (build hooks)
+        📄 example_native_library.c
+        📄 example_native_library.h
+    📁  test                          // test your assets here
+        📄 example_test.dart
+```
+
+## Hooks
+
+Currently, the build hook and link hook are available,
+but more are planned. To learn more about these hooks,
+see the following sections.
+
+### Build hooks
+
+A build hook builds assets that you want
+to run during your project's build process. This is useful
+for tasks that need to happen before your main
+application code is compiled, such as:
+
+* Compiling C, C++, Rust, or other native code into a
+  dynamic library.
+* Downloading or generating other assets required at
+  runtime.
+
+Use the [`build`][] function to create a build hook.
+Build hooks are stored in the `hook/build.dart` directory
+and create code assets in the `build/` directory.
+
+[`build`]: {{site.pub-api}}/hooks/0.19.5/hooks/build.html
+
+### Link hooks
+
+A link hook shrinks assets to only the
+required parts based on Dart AOT tree-shaking information.
+
+Use the [`link`][] function to create a build hook.
+Link hooks are stored in the `hook/link.dart` directory
+and create code assets and data assets in the
+`build/` directory.
+
+[`link`]: {{site.pub-api}}/hooks/0.19.5/hooks/link.html
+
+## Assets
+
+Assets are the files output from a hook that need to be
+bundled in a Dart application. Assets can be accessed
+at runtime from the Dart code. Currently, the Dart SDK can
+use the `CodeAsset` and `DataAsset` asset types.
+
+### CodeAsset type
+
+A [`CodeAsset`][] represents a code asset. A code asset is
+a dynamic library compiled from a language other than
+Dart, such as C, C++, Rust, or Go. `CodeAsset` is part of
+the `code_asset` package and are called with `dart:ffi` from
+the Dart code at runtime using the `@Native` annotation.
+A code asset can be used with build and link hooks.
+
+[`CodeAsset`]: {{site.pub-api}}/code_assets/latest/code_assets/CodeAsset-class.html
+
+### DataAsset type
+
+A [`DataAsset`][] represents a data asset in bytes format.
+It can be accessed by Dart at runtime and used in
+Dart standalone packages. `DataAsset` is part of the
+`data_asset` package and can be used with link hooks.
+
+[`DataAsset`]: {{site.pub-api}}/data_assets/latest/data_assets/DataAsset-class.html
+
+## Use a hook
+
+To setup a hook, use a hook to generate assets, and use
```

Comment:
What did you want to say here?

To setup a hook use a hook sounds weird.

To add assets to your project, use ...